G_CONST_RETURN is deprecated in glib 2.30 so remove to to avoid
build failures. Also resolve other deprecation issues.

RP 2011/10/12

Upstream-Status: Pending

Index: git/libedataserver/e-data-server-util.c
===================================================================
--- git.orig/libedataserver/e-data-server-util.c	2011-10-12 01:42:40.622045788 +0100
+++ git/libedataserver/e-data-server-util.c	2011-10-12 01:48:16.732045664 +0100
@@ -165,7 +165,8 @@
 static gunichar
 stripped_char (gunichar ch)
 {
-        gunichar *decomp, retval;
+        gunichar decomp[4];
+        gunichar retval;
         GUnicodeType utype;
         gsize dlen;
 
@@ -175,7 +176,7 @@
         case G_UNICODE_CONTROL:
         case G_UNICODE_FORMAT:
         case G_UNICODE_UNASSIGNED:
-        case G_UNICODE_COMBINING_MARK:
+        case G_UNICODE_SPACING_MARK:
                 /* Ignore those */
                 return 0;
                break;
@@ -183,7 +184,7 @@
                 /* Convert to lowercase, fall through */
                 ch = g_unichar_tolower (ch);
         case G_UNICODE_LOWERCASE_LETTER:
-                if ((decomp = g_unicode_canonical_decomposition (ch, &dlen))) {
+                if ((dlen = g_unichar_fully_decompose (ch, FALSE, decomp, 4))) {
                         retval = decomp[0];
                         g_free (decomp);
                         return retval;
Index: git/addressbook/libedata-book/e-data-book-factory.c
===================================================================
--- git.orig/addressbook/libedata-book/e-data-book-factory.c	2011-10-12 01:52:25.692045500 +0100
+++ git/addressbook/libedata-book/e-data-book-factory.c	2011-10-12 01:53:31.862045707 +0100
@@ -238,7 +238,7 @@
 	return g_strdup_printf (
 		"/org/gnome/evolution/dataserver/addressbook/%d/%u",
 		getpid (),
-		g_atomic_int_exchange_and_add (&counter, 1));
+		g_atomic_int_add (&counter, 1));
 }
 
 static void
Index: git/addressbook/libedata-book/e-data-book.c
===================================================================
--- git.orig/addressbook/libedata-book/e-data-book.c	2011-10-12 01:54:40.752045465 +0100
+++ git/addressbook/libedata-book/e-data-book.c	2011-10-12 01:55:16.012045897 +0100
@@ -553,7 +553,7 @@
 
 	return g_strdup_printf ("/org/gnome/evolution/dataserver/addressbook/BookView/%d/%d",
 				getpid (),
-				g_atomic_int_exchange_and_add ((int*)&counter, 1));
+				g_atomic_int_add ((int*)&counter, 1));
 }
 
 static void
Index: git/calendar/libedata-cal/e-data-cal-factory.c
===================================================================
--- git.orig/calendar/libedata-cal/e-data-cal-factory.c	2011-10-12 01:57:56.612045081 +0100
+++ git/calendar/libedata-cal/e-data-cal-factory.c	2011-10-12 01:57:58.832045211 +0100
@@ -227,7 +227,7 @@
 	return g_strdup_printf (
 		"/org/gnome/evolution/dataserver/calendar/%d/%u",
 		getpid (),
-		g_atomic_int_exchange_and_add (&counter, 1));
+		g_atomic_int_add (&counter, 1));
 }
 
 static void
